edb/pgsql/compiler/context.py

```diff
@@ -304,6 +304,10 @@ class CompilerContextLevel(compiler.ContextLevel):
     #: needed by DML.
     shapes_needed_by_dml: Set[irast.Set]
 
+    #: Aggregated output of detached SQL parameters (position integer) and
+    #: their SQL types (qualified names tuple)
+    detached_params: Dict[int, Tuple[str, ...]]
+
```

Comment:
What is a detached parameter?

Reply:
We normally compile in a way that the types of input parameters are placed inline, like

```sql
... where users.name = $3::text
```

but in functions, we'd have to define the type of parameters in the function signature, like:

```sql
create function find_user(name: text)
...
begin
    ... where users.name = $name
...
```

`detached_params` is just my way to find those types of parameters.